### PR TITLE
Refactor: update StatTracker and remove unnecessary comments

### LINKED ISSUE #37: Cleanup / Refactor

### PR DESCRIPTION
**Pull request by westonio**
## Describe your changes in detail
- Removes all unnecessary comments throughout files (commented out code no longer needed)
- Adds a helper method for most/fewest tackles methods, and a relevant test (named badly - sorry)
- Refactors a few methods

## Checklist before requesting a review
- [ x ] I have performed a self-review of my code
- [ x ] I have added thorough tests
- [ x ] I AM MERGING WITH Westonio/futbol

## Anything else noteworthy? (E.g. note anything that might need refactoring or special attention later)
All local tests and the spec harness tests pass after these updates :)

### CODE REVIEW

**Review by topher-nullset — APPROVED** [review]
Looks really good, nice cleanup!

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -177,70 +153,71 @@ def worst_coach
...
-    [team, tackles.map { |game| game.tackles}]
+    games_in_season = season_games(season_id)
+    games_by_team = games_in_season.group_by { |team| team.team_id }
+    team_tackles = games_by_team.map do |team, tackles|
+      [team, tackles.map { |game| game.tackles}]
     end
     tackle_totals = team_tackles.map {|team, tackles| [team, tackles.sum]}.to_h
     highest_total = tackle_totals.min_by { |team, tackles| tackles}
     @teams.find { |team| team.team_id == highest_total.first}.team_name
   end
```
Comment (pcbennett108) [inline lib/stat_tracker.rb:240]: Prime candidate for refactor for sure.
I think that what I'm working on right now will also be able to benefit from this.

**Review by pcbennett108 — COMMENTED** [review]
Thanks for taking out some of these extra comments. I had forgotten about them.

**Comment by westonio** [comment]
Updated the winningest and worst coach methods to use the helper method .season_games

**Pull request merged**